From c6820de05abf3f8aee75b40254c883197ba09712 Mon Sep 17 00:00:00 2001
From: Andy Wilkinson <awilkinson@pivotal.io>
Date: Tue, 28 Feb 2017 10:25:06 +0000
Subject: [PATCH] Import auto-configuration so that it is ordered correctly

When imported an auto-configuration class, @ImportAutoConfiguration
should be used. This ensures that auto-configuration classes are
automatically ordered correctly relative to each other. It also
ensures that the auto-configuration classes are considered after any
user-provided configuration which ensures that any bean conditions
will work correctly
---
 .../org/springframework/integration/dsl/test/amqp/AmqpTests.java    | 6 +++---
 .../java/org/springframework/integration/dsl/test/ftp/FtpTests.java | 6 ++++--
 .../org/springframework/integration/dsl/test/http/HttpTests.java    | 6 +++---
 .../org/springframework/integration/dsl/test/jdbc/JdbcTests.java    | 6 +++---
 .../java/org/springframework/integration/dsl/test/jms/JmsTests.java | 6 +++---
 .../java/org/springframework/integration/dsl/test/jpa/JpaTests.java | 6 +++---
 .../springframework/integration/dsl/test/mongodb/MongoDbTests.java  | 6 +++---
 .../org/springframework/integration/dsl/test/sftp/SftpTests.java    | 6 ++++--
 8 files changed, 26 insertions(+), 22 deletions(-)

diff --git a/src/test/java/org/springframework/integration/dsl/test/amqp/AmqpTests.java b/src/test/java/org/springframework/integration/dsl/test/amqp/AmqpTests.java
index 5e54804..80560b6 100644
--- a/src/test/java/org/springframework/integration/dsl/test/amqp/AmqpTests.java
+++ b/src/test/java/org/springframework/integration/dsl/test/amqp/AmqpTests.java
@@ -1,5 +1,5 @@
 /*
- * Copyright 2014-2015 the original author or authors.
+ * Copyright 2014-2017 the original author or authors.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -33,10 +33,10 @@ import org.springframework.amqp.rabbit.connection.ConnectionFactory;
 import org.springframework.amqp.rabbit.core.RabbitTemplate;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.boot.autoconfigure.ImportAutoConfiguration;
 import org.springframework.boot.autoconfigure.amqp.RabbitAutoConfiguration;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
-import org.springframework.context.annotation.Import;
 import org.springframework.integration.amqp.channel.AbstractAmqpChannel;
 import org.springframework.integration.amqp.inbound.AmqpInboundGateway;
 import org.springframework.integration.amqp.support.AmqpHeaderMapper;
@@ -170,7 +170,7 @@ public class AmqpTests {
 
 	@Configuration
 	@EnableIntegration
-	@Import(RabbitAutoConfiguration.class)
+	@ImportAutoConfiguration(RabbitAutoConfiguration.class)
 	public static class ContextConfiguration {
 
 		@Bean
diff --git a/src/test/java/org/springframework/integration/dsl/test/ftp/FtpTests.java b/src/test/java/org/springframework/integration/dsl/test/ftp/FtpTests.java
index 119ec54..2c0b076 100644
--- a/src/test/java/org/springframework/integration/dsl/test/ftp/FtpTests.java
+++ b/src/test/java/org/springframework/integration/dsl/test/ftp/FtpTests.java
@@ -1,5 +1,5 @@
 /*
- * Copyright 2014-2016 the original author or authors.
+ * Copyright 2014-2017 the original author or authors.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -41,6 +41,7 @@ import org.junit.runner.RunWith;
 
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.boot.autoconfigure.ImportAutoConfiguration;
 import org.springframework.boot.autoconfigure.integration.IntegrationAutoConfiguration;
 import org.springframework.boot.autoconfigure.jmx.JmxAutoConfiguration;
 import org.springframework.context.annotation.Bean;
@@ -183,7 +184,8 @@ public class FtpTests {
 	}
 
 	@Configuration
-	@Import({TestFtpServer.class, JmxAutoConfiguration.class, IntegrationAutoConfiguration.class})
+	@Import(TestFtpServer.class)
+	@ImportAutoConfiguration({JmxAutoConfiguration.class, IntegrationAutoConfiguration.class})
 	@IntegrationComponentScan
 	public static class ContextConfiguration {
 
diff --git a/src/test/java/org/springframework/integration/dsl/test/http/HttpTests.java b/src/test/java/org/springframework/integration/dsl/test/http/HttpTests.java
index 54c1acf..ef59ca8 100644
--- a/src/test/java/org/springframework/integration/dsl/test/http/HttpTests.java
+++ b/src/test/java/org/springframework/integration/dsl/test/http/HttpTests.java
@@ -1,5 +1,5 @@
 /*
- * Copyright 2015-2016 the original author or authors.
+ * Copyright 2015-2017 the original author or authors.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -25,6 +25,7 @@ import org.junit.Test;
 import org.junit.runner.RunWith;
 
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.ImportAutoConfiguration;
 import org.springframework.boot.autoconfigure.PropertyPlaceholderAutoConfiguration;
 import org.springframework.boot.autoconfigure.security.SecurityAutoConfiguration;
 import org.springframework.boot.autoconfigure.security.SecurityProperties;
@@ -36,7 +37,6 @@ import org.springframework.boot.test.context.SpringBootTest;
 import org.springframework.boot.test.web.client.TestRestTemplate;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
-import org.springframework.context.annotation.Import;
 import org.springframework.core.annotation.Order;
 import org.springframework.core.env.Environment;
 import org.springframework.integration.channel.DirectChannel;
@@ -79,7 +79,7 @@ public class HttpTests {
 
 
 	@Configuration
-	@Import({PropertyPlaceholderAutoConfiguration.class, ServerPropertiesAutoConfiguration.class,
+	@ImportAutoConfiguration({PropertyPlaceholderAutoConfiguration.class, ServerPropertiesAutoConfiguration.class,
 			EmbeddedServletContainerAutoConfiguration.class, DispatcherServletAutoConfiguration.class,
 			SecurityAutoConfiguration.class })
 	@EnableIntegration
diff --git a/src/test/java/org/springframework/integration/dsl/test/jdbc/JdbcTests.java b/src/test/java/org/springframework/integration/dsl/test/jdbc/JdbcTests.java
index 7599c7c..a0311ee 100644
--- a/src/test/java/org/springframework/integration/dsl/test/jdbc/JdbcTests.java
+++ b/src/test/java/org/springframework/integration/dsl/test/jdbc/JdbcTests.java
@@ -1,5 +1,5 @@
 /*
- * Copyright 2014-2016 the original author or authors.
+ * Copyright 2014-2017 the original author or authors.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -31,11 +31,11 @@ import org.junit.runner.RunWith;
 
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.boot.autoconfigure.ImportAutoConfiguration;
 import org.springframework.boot.autoconfigure.jdbc.DataSourceAutoConfiguration;
 import org.springframework.boot.test.context.SpringBootTest;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
-import org.springframework.context.annotation.Import;
 import org.springframework.integration.config.EnableIntegration;
 import org.springframework.integration.dsl.IntegrationFlow;
 import org.springframework.jdbc.InvalidResultSetAccessException;
@@ -76,7 +76,7 @@ public class JdbcTests {
 	}
 
 	@Configuration
-	@Import(DataSourceAutoConfiguration.class)
+	@ImportAutoConfiguration(DataSourceAutoConfiguration.class)
 	@EnableIntegration
 	public static class ContextConfiguration {
 
diff --git a/src/test/java/org/springframework/integration/dsl/test/jms/JmsTests.java b/src/test/java/org/springframework/integration/dsl/test/jms/JmsTests.java
index ad535d8..beb7cf5 100644
--- a/src/test/java/org/springframework/integration/dsl/test/jms/JmsTests.java
+++ b/src/test/java/org/springframework/integration/dsl/test/jms/JmsTests.java
@@ -1,5 +1,5 @@
 /*
- * Copyright 2014-2016 the original author or authors.
+ * Copyright 2014-2017 the original author or authors.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -39,13 +39,13 @@ import org.junit.runner.RunWith;
 import org.springframework.beans.factory.ListableBeanFactory;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.boot.autoconfigure.ImportAutoConfiguration;
 import org.springframework.boot.autoconfigure.integration.IntegrationAutoConfiguration;
 import org.springframework.boot.autoconfigure.jms.activemq.ActiveMQAutoConfiguration;
 import org.springframework.boot.autoconfigure.jmx.JmxAutoConfiguration;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.ComponentScan;
 import org.springframework.context.annotation.Configuration;
-import org.springframework.context.annotation.Import;
 import org.springframework.integration.annotation.InboundChannelAdapter;
 import org.springframework.integration.annotation.IntegrationComponentScan;
 import org.springframework.integration.annotation.MessagingGateway;
@@ -232,7 +232,7 @@ public class JmsTests {
 	}
 
 	@Configuration
-	@Import({ ActiveMQAutoConfiguration.class, JmxAutoConfiguration.class, IntegrationAutoConfiguration.class })
+	@ImportAutoConfiguration({ ActiveMQAutoConfiguration.class, JmxAutoConfiguration.class, IntegrationAutoConfiguration.class })
 	@IntegrationComponentScan
 	@ComponentScan
 	public static class ContextConfiguration {
diff --git a/src/test/java/org/springframework/integration/dsl/test/jpa/JpaTests.java b/src/test/java/org/springframework/integration/dsl/test/jpa/JpaTests.java
index ca11e78..492085e 100644
--- a/src/test/java/org/springframework/integration/dsl/test/jpa/JpaTests.java
+++ b/src/test/java/org/springframework/integration/dsl/test/jpa/JpaTests.java
@@ -1,5 +1,5 @@
 /*
- * Copyright 2016 the original author or authors.
+ * Copyright 2016-2017 the original author or authors.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -35,6 +35,7 @@ import org.junit.runner.RunWith;
 
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.boot.autoconfigure.ImportAutoConfiguration;
 import org.springframework.boot.autoconfigure.domain.EntityScan;
 import org.springframework.boot.autoconfigure.integration.IntegrationAutoConfiguration;
 import org.springframework.boot.autoconfigure.jdbc.DataSourceAutoConfiguration;
@@ -42,7 +43,6 @@ import org.springframework.boot.autoconfigure.orm.jpa.HibernateJpaAutoConfigurat
 import org.springframework.boot.test.context.SpringBootTest;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
-import org.springframework.context.annotation.Import;
 import org.springframework.integration.dsl.IntegrationFlow;
 import org.springframework.integration.dsl.IntegrationFlows;
 import org.springframework.integration.dsl.jpa.Jpa;
@@ -168,7 +168,7 @@ public class JpaTests {
 
 
 	@Configuration
-	@Import({ DataSourceAutoConfiguration.class, HibernateJpaAutoConfiguration.class,
+	@ImportAutoConfiguration({ DataSourceAutoConfiguration.class, HibernateJpaAutoConfiguration.class,
 			IntegrationAutoConfiguration.class })
 	@EntityScan(basePackageClasses = StudentDomain.class)
 	public static class ContextConfiguration {
diff --git a/src/test/java/org/springframework/integration/dsl/test/mongodb/MongoDbTests.java b/src/test/java/org/springframework/integration/dsl/test/mongodb/MongoDbTests.java
index e70dd58..0494728 100644
--- a/src/test/java/org/springframework/integration/dsl/test/mongodb/MongoDbTests.java
+++ b/src/test/java/org/springframework/integration/dsl/test/mongodb/MongoDbTests.java
@@ -1,5 +1,5 @@
 /*
- * Copyright 2014-2016 the original author or authors.
+ * Copyright 2014-2017 the original author or authors.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -27,13 +27,13 @@ import org.junit.runner.RunWith;
 
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.boot.autoconfigure.ImportAutoConfiguration;
 import org.springframework.boot.autoconfigure.data.mongo.MongoDataAutoConfiguration;
 import org.springframework.boot.autoconfigure.mongo.MongoAutoConfiguration;
 import org.springframework.boot.autoconfigure.mongo.embedded.EmbeddedMongoAutoConfiguration;
 import org.springframework.boot.test.context.SpringBootTest;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
-import org.springframework.context.annotation.Import;
 import org.springframework.core.convert.converter.Converter;
 import org.springframework.data.mongodb.MongoDbFactory;
 import org.springframework.data.mongodb.config.EnableMongoAuditing;
@@ -144,7 +144,7 @@ public class MongoDbTests {
 
 
 	@Configuration
-	@Import({EmbeddedMongoAutoConfiguration.class, MongoAutoConfiguration.class, MongoDataAutoConfiguration.class})
+	@ImportAutoConfiguration({EmbeddedMongoAutoConfiguration.class, MongoAutoConfiguration.class, MongoDataAutoConfiguration.class})
 	@EnableIntegration
 	@IntegrationComponentScan
 	@EnableMongoAuditing
diff --git a/src/test/java/org/springframework/integration/dsl/test/sftp/SftpTests.java b/src/test/java/org/springframework/integration/dsl/test/sftp/SftpTests.java
index bc3f65e..2c54639 100644
--- a/src/test/java/org/springframework/integration/dsl/test/sftp/SftpTests.java
+++ b/src/test/java/org/springframework/integration/dsl/test/sftp/SftpTests.java
@@ -1,5 +1,5 @@
 /*
- * Copyright 2014-2016 the original author or authors.
+ * Copyright 2014-2017 the original author or authors.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -41,6 +41,7 @@ import org.junit.runner.RunWith;
 
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.boot.autoconfigure.ImportAutoConfiguration;
 import org.springframework.boot.autoconfigure.integration.IntegrationAutoConfiguration;
 import org.springframework.boot.autoconfigure.jmx.JmxAutoConfiguration;
 import org.springframework.context.annotation.Bean;
@@ -206,7 +207,8 @@ public class SftpTests {
 	}
 
 	@Configuration
-	@Import({TestSftpServer.class, JmxAutoConfiguration.class, IntegrationAutoConfiguration.class})
+	@Import(TestSftpServer.class)
+	@ImportAutoConfiguration({JmxAutoConfiguration.class, IntegrationAutoConfiguration.class})
 	@IntegrationComponentScan
 	public static class ContextConfiguration {
 
-- 
2.11.0

